Makefile.common

```diff
@@ -108,7 +108,15 @@ CXXFLAGS += -fmacro-prefix-map=$(dir $(TOPDIR))=
 # garbage-collect unused code/data
 # NOTE: exported to ports also as it reduces binaries size greatly
 CFLAGS += -ffunction-sections -fdata-sections
-LDFLAGS += $(LDFLAGS_PREFIX)--gc-sections
+ifeq ($(APPLE_LD), 1)
+  LDFLAGS += $(LDFLAGS_PREFIX)-dead_strip
+  LDFLAG1 = $(LDFLAGS_PREFIX)-force_load
```

Comment:
regarding the `-force_load` - I'm not sure (no way to test it), but I think the current solution is invalid. The option `-force_load` takes always one library as a parameter, there is no grouping and no end-parameter here. I suspect `LDFLAG2` has invalid value (does nothing, introduces confusion).

We either need to use `-all_load` (to load all libraries) or create one param per library (to be tested on Darwin).

Please confirm that you understand why `--whole-archive` param is needed (described in the comment) to test against this particular use case.

Reply:
Yes, I understand the need for `--whole-archive` and, as you say,` -all_load` can be used as a substitute on the Darwin architecture. `-force_load` takes only one library. I tested the behavior with `all_load `on Darwin, by creating various constructors in different host-utils, and the result was always correct.